src/cli/tls_http_server.cpp

```diff
@@ -43,305 +49,51 @@ namespace Botan_CLI {
    private:
-      size_t m_max_clients;
-      std::atomic<size_t> m_clients_serviced;
-};
-
-/*
-* This is an incomplete and highly buggy HTTP request parser. It is just
-* barely sufficient to handle a GET request sent by a browser.
-*/
-class HTTP_Parser final {
-   public:
-      class Request {
-         public:
-            const std::string& verb() const { return m_verb; }
-
-            const std::string& location() const { return m_location; }
-
-            const std::map<std::string, std::string>& headers() const { return m_headers; }
-
-            Request(const std::string& verb,
-                    const std::string& location,
-                    const std::map<std::string, std::string>& headers) :
-                  m_verb(verb), m_location(location), m_headers(headers) {}
-
-         private:
-            std::string m_verb;
-            std::string m_location;
-            std::map<std::string, std::string> m_headers;
-      };
-
-      class Callbacks {
-         public:
-            virtual void handle_http_request(const Request& request) = 0;
-
-            virtual ~Callbacks() = default;
-
-            Callbacks() = default;
-
-            Callbacks(const Callbacks& other) = delete;
-            Callbacks(Callbacks&& other) = delete;
-            Callbacks& operator=(const Callbacks& other) = delete;
-            Callbacks& operator=(Callbacks&&) = delete;
-      };
-
-      HTTP_Parser(Callbacks& cb) : m_cb(cb) {}
-
-      void consume_input(std::span<const uint8_t> buf) {
-         m_req_buf.append(reinterpret_cast<const char*>(buf.data()), buf.size());
-
-         std::istringstream strm(m_req_buf);
+      auto timestamp() const {
+         const auto t = std::time(nullptr);
+         struct tm tm;
 
-         std::string http_version;
-         std::string verb;
-         std::string location;
-         std::map<std::string, std::string> headers;
-
-         strm >> verb >> location >> http_version;
-
-         if(verb.empty() || location.empty()) {
-            return;
-         }
-
-         while(true) {
-            std::string header_line;
-            std::getline(strm, header_line);
-
-            if(header_line == "\r") {
-               continue;
-            }
-
-            auto delim = header_line.find(": ");
-            if(delim == std::string::npos) {
-               break;
-            }
-
-            const std::string hdr_name = header_line.substr(0, delim);
-            const std::string hdr_val = header_line.substr(delim + 2, std::string::npos);
-
-            headers[hdr_name] = hdr_val;
-
-            if(headers.size() > 1024) {
-               throw Botan::Invalid_Argument("Too many HTTP headers sent in request");
-            }
-         }
+   #if defined(BOTAN_BUILD_COMPILER_IS_MSVC) || defined(BOTAN_TARGET_OS_IS_MINGW) || \
+      defined(BOTAN_TARGET_OS_IS_CYGWIN) || defined(BOTAN_TARGET_OS_IS_WINDOWS)
+         localtime_s(&tm, &t);
+   #else
+         localtime_r(&t, &tm);
```

Comment:
Can we use `std::chrono::local_time` here now?

Reply:
It seems that the ["Calendar and timezone" features](https://en.cppreference.com/w/cpp/compiler_support#C.2B.2B20_library_features) aren't implemented in the STL, yet. :(

Reply:
:sob: